src/libstore/derivations.cc

```diff
@@ -8,6 +8,32 @@
 
 namespace nix {
 
+bool derivationIsCA(DerivationType dt) {
+    switch (dt) {
+    case DerivationType::Regular: return false;
+    case DerivationType::CAFixed: return true;
+    };
+    // Since enums can have non-variant values, but making a `default:` would
+    // disable exhaustiveness warnings.
+    abort();
```

Comment:
Could you change this to `assert(false)`? That way, if we ever hit this assertion, we get an error message with location info.

Reply:
That's changed now, thanks